docs/en/stack/ml/df-analytics/ml-trained-models.asciidoc

```diff
@@ -13,168 +13,146 @@ information about this process, see <<ml-supervised-workflow>> and
+{ref}/get-trained-models.html[Get Trained Model API] to export the entire model
+definition and save it to a JSON file.
++
+--
+[source, bash]
 --------------------------------------------------
-import json
-from elasticsearch import Elasticsearch
-from elasticsearch.client.ml import MlClient
-es_client = Elasticsearch(args.es, http_auth=(username, password), use_ssl=True, timeout=60)
-ml_client = MlClient(es_client)
-with open(filename, 'r') as handle:
-  compressed_model = json.loads(handle.read())
-for field in ('version', 'create_time', 'estimated_heap_memory_usage_bytes', 'estimated_operations', 'license_level', 'id','created_by'):
-  if field in compressed_model:
-    del compressed_model[field]
-ml_client.put_trained_model(model_id=model_id, body=compressed_model)
+curl -u elastic:changeme \
+  -X GET "http://localhost:9200/_ml/trained_models/flights1-1607953694065?exclude_generated=true&include=definition&decompress_definition=false" \
+    | jq '.trained_model_configs[0] | del(.model_id)' \
+    > flights1.json
 --------------------------------------------------
 // NOTCONSOLE
 
+A few observations:
 
-You can achieve the same by running a bash script. Populate the environment 
-variables:
-
-`ES_ADDRESS="https://username:password@elasticsearch-address"`
+* Exporting models requires using `curl` or a similar tool that can *stream*
+the model over HTTP into a file. Using the https://www.elastic.co/guide/en/
+kibana/current/console-kibana.html[Kibana Dev Tools console] will cause the
+browser to become unresponsive due to the size of exported models.
 
-`MODEL="my_model_name"`
+* Note the query parameters that are used during export. These are necessary to
+export the model in a way that it can later be imported again and used for
+inference.
 
+* We need to unnest the JSON object by one level to extract just the model
+definition. We also want to remove the existing model ID in order to not have
+ID collisions when we import again. We do these steps using `jq` inline but
+alternatively it can be done to the resulting JSON file after downloading using
+`jq` or other tools.
+--
 
-Then run the script:
-
+3. Import the saved model using `curl` to upload the JSON file to the {ref}/
+create-trained-models.html[Created Trained Model API]. When we specify the URL,
+we can also set the model ID to something new using the last path part of the
+URL.
++
+--
 [source, bash]
 --------------------------------------------------
-curl -H 'Content-Type: application/json' -XPUT "$ES_ADDRESS/_ml/inference/$MODEL" -d@$MODEL.json
+curl -u elastic:changeme \
+  -H 'Content-Type: application/json' \
+  -X PUT "http://localhost:9200/_ml/trained_models/flights1-imported" \
+  --data-binary @flights1.json
 --------------------------------------------------
 // NOTCONSOLE
+--
+
+[NOTE]
+--
+* Models exported from the {ref}/get-trained-models.html[Get Trained Model API]
+are limited in size by the https://www.elastic.co/guide/en/elasticsearch/
```

Comment:
Not sure why this link is fully qualified. I think it should use {ref} same as others...
```suggestion
are limited in size by the
```

Reply:
Sorry, I didn't have the right {ref}'s for some links.